lib/tasks/test.rake

```diff
@@ -9,86 +9,197 @@ namespace :test do
-      !file.end_with?('xml') &&
-      !File.directory?(file)
+  def call_hets_provers(file, subdir)
+    puts "Calling hets/provers on #{file.inspect}"
+    options = 'format=json'
+    escaped_iri = Rack::Utils.escape_path("file://#{absolute_filepath(file)}")
+    hets_iri = "#{HETS_BASE_IRI}/provers/#{escaped_iri}#{HETS_API_OPTIONS}"
+    hets_iri << "?#{options}"
+
+    FileUtils.rm_f(recorded_file(subdir, file))
+    VCR.use_cassette(cassette_path_in_fixtures(subdir, file)) do
+      Net::HTTP.get_response(URI(hets_iri))
     end
   end
 
-  def prove_files
-    globbed_files = Dir.glob('spec/fixtures/ontologies/prove/**/*')
-    globbed_files.select do |file|
-      !file.end_with?('.proof.json') && !File.directory?(file)
+  def call_hets_prove(file, subdir)
+    puts "Calling hets/prove on #{file.inspect}"
+    escaped_iri = Rack::Utils.escape_path("file://#{absolute_filepath(file)}")
+    hets_iri = "#{HETS_BASE_IRI}/prove/#{escaped_iri}#{HETS_API_OPTIONS}"
+    header = {'Content-Type' => 'application/json'}
+    data = {format: 'json', include: 'true'}
+
+    FileUtils.rm_f(recorded_file(subdir, file))
+    VCR.use_cassette(cassette_path_in_fixtures(subdir, file)) do
+      uri = URI(hets_iri)
+      Net::HTTP.start(uri.hostname, uri.port) do |http|
+        http.request_post(uri, data.to_json, header)
+      end
+    end
+  end
+
+  def freshen_ontology_fixtures
+    on_outdated_cassettes(ontology_files, 'dg') do |file, subdir|
+      call_hets_dg(file, subdir)
     end
   end
 
-  def prove_with_hets(file)
-    puts "Calling hets prover for #{file}"
+  def freshen_provers_fixtures
+    on_outdated_cassettes(ontology_files, 'provers') do |file, subdir|
+      call_hets_provers(file, subdir)
+    end
+  end
+
+  def freshen_proof_fixtures
+    on_outdated_cassettes(prove_files, 'prove') do |file, subdir|
+      call_hets_prove(file, subdir)
+    end
+  end
+
+  def setup_vcr
+    require 'vcr'
+    VCR.configure do |c|
+      c.cassette_library_dir = 'spec/fixtures/vcr'
+      c.hook_into :webmock
+    end
+  end
 
-    absolute_filepath = Rails.root.join(file)
-    escaped_iri = Rack::Utils.escape_path("file://#{absolute_filepath}")
-    command = %w(curl -s -X POST)
-    command += ['-H', %('Content-Type: application/json')]
-    command += ['-d', %('{"format": "json"}')]
-    command << "http://localhost:8000/prove/#{escaped_iri}"
-    command = command.join(' ')
+  def port_open?(ip, port, seconds=1)
+    require 'socket'
+    require 'timeout'
+    Timeout::timeout(seconds) do
+      begin
+        TCPSocket.new(ip, port).close
+        true
+      rescue Errno::ECONNREFUSED, Errno::EHOSTUNREACH
+        false
+      end
+    end
+  rescue Timeout::Error
+    false
+  end
 
-    filename = File.basename(file).split('.')[0..-2].join('.')
-    target_path = Rails.root.join('spec/fixtures/ontologies/hets-out/prove', filename)
+  def with_running_hets(&block)
+    hets_was_already_running = !port_open?('127.0.0.1', 8000)
```

Comment:
Right, thanks. This is a bad variable name. I fixed it in f46f2125b8cdf6d8240c1b85b438f1fbc51c2da8.